src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -170,4 +280,253 @@ public void parseTags_collectionWithValidTags_returnsTagSet() throws Exception {
+    }
+
+    @Test
+    public void parseTime_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseTime((String) null));
+    }
+
+    @Test
+    public void parseTime_invalidValue_throwsParseException() {
+        assertThrows(ParseException.class, () -> ParserUtil.parseTime(INVALID_TIME));
+    }
+
+    @Test
+    public void parseTime_validValueWithoutWhitespace_returnsTime() throws Exception {
+        Time expectedTime1 = new Time(VALID_TIME_1);
+        Time expectedTime2 = new Time(VALID_TIME_2);
+
+        assertEquals(expectedTime1, ParserUtil.parseTime(VALID_TIME_1));
+        assertEquals(expectedTime2, ParserUtil.parseTime(VALID_TIME_2));
+    }
+
+    @Test
+    public void parseTime_validValueWithWhitespace_returnsTrimmedTime() throws Exception {
+        String websiteWithWhitespace1 = WHITESPACE + VALID_TIME_1 + WHITESPACE;
+        String websiteWithWhitespace2 = WHITESPACE + VALID_TIME_2 + WHITESPACE;
+        Time expectedTime1 = new Time(VALID_TIME_1);
+        Time expectedTime2 = new Time(VALID_TIME_2);
+
+        assertEquals(expectedTime1, ParserUtil.parseTime(websiteWithWhitespace1));
+        assertEquals(expectedTime2, ParserUtil.parseTime(websiteWithWhitespace2));
+    }
+
+    @Test
+    public void parseReditItemIndices_null_returnOptionalEmpty() throws Exception {
+        assertEquals(ItemIndicesBuilder.empty(), ParserUtil.parseReditItemIndices(null));
+    }
+
+    @Test
+    public void parseReditItemIndices_emptyString_returnOptionalOfEmptyArray() throws Exception {
+        assertEquals(new ItemIndicesBuilder().build(), ParserUtil.parseReditItemIndices(""));
+    }
+
+    @Test
+    public void parseReditItemIndices_invalidIndices_returnOptionalOfIndices() throws Exception {
+        assertThrows(ParseException.class, () -> ParserUtil.parseReditItemIndices("2 a"));
+        assertThrows(ParseException.class, () -> ParserUtil.parseReditItemIndices("a"));
+        assertThrows(ParseException.class, () -> ParserUtil.parseReditItemIndices("0"));
+        assertThrows(ParseException.class, () -> ParserUtil.parseReditItemIndices("-1"));
+    }
+
+    @Test
+    public void parseReditItemIndices_validIndicesWithoutWhitespace_returnOptionalOfIndices() throws Exception {
+        ItemIndicesBuilder expectedIndices1 = new ItemIndicesBuilder().add(2).add(1).add(3).add(5);
+        assertEquals(expectedIndices1.build(), ParserUtil.parseReditItemIndices("2 1 3 5"));
+
+        ItemIndicesBuilder expectedIndices2 = new ItemIndicesBuilder().add(2);
+        assertEquals(expectedIndices2.build(), ParserUtil.parseReditItemIndices("2"));
+    }
+
+
+    @Test
+    public void parseDisplayPicture_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseDisplayPicture((String) null));
+    }
+
+    // TODO: I don't know an invalid file path. Can someone help?
```

Comment:
I checked online with our current regex for dp file path, at it kinda throws a warning, time to edit DP regex for something safer?

Reply:
Will update the regex in another PR. Scared it will break something and make this PR unmerge-able